components/backup-stream/src/router.rs

```diff
@@ -2413,4 +2424,77 @@ mod tests {
         let r = cfg_manager.dispatch(changed);
         assert!(r.is_err());
     }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+    async fn test_flush_on_events_race() -> Result<()> {
+        let (tx, _rx) = dummy_scheduler();
+        let tmp = std::env::temp_dir().join(format!("{}", uuid::Uuid::new_v4()));
+        let router = Arc::new(RouterInner::new(
+            tx,
+            Config {
+                prefix: tmp.clone(),
+                // disable auto flush.
+                temp_file_size_limit: 1000,
+                temp_file_memory_quota: 2,
+                max_flush_interval: Duration::from_secs(300),
+            },
+        ));
+
+        let (task, _path) = task("race".to_owned()).await?;
+        must_register_table(router.as_ref(), task, 1).await;
+        router
+            .must_mut_task_info("race", |i| {
+                i.storage = Arc::new(NoopStorage::default());
+            })
+            .await;
+        let mut b = KvEventsBuilder::new(42, 0);
+        b.put_table(CF_DEFAULT, 1, "k1".as_bytes(), "v1".as_bytes());
```

Comment:
Perhaps we can provide binary string directly. (Not checked, if this won't compile, also try `&b"k1"` or ignore this.)
```suggestion
        b.put_table(CF_DEFAULT, 1, b"k1", b"v1");
```